trans.js

```diff
@@ -0,0 +1,264 @@
+const JSZip = require('jszip');
+const os = require('os');
+const program = require('commander');
+
+const CROWDIN_PROJECT = 'wire-cs';
+const DEFAULT_LOCALE_SOURCE = path.join(__dirname, 'electron', 'src', 'locale');
+const DEFAULT_LOCALE_STRINGS = path.join(__dirname, 'electron', 'src', 'locale', 'strings.ts');
+const TEMP_DIR = fs.mkdtempSync(path.join(os.tmpdir(), 'wd-'));
+const TEMP_ZIP_FILE = path.join(TEMP_DIR, 'translations.zip');
+const NO_API_KEY_ERROR_MESSAGE =
+  '\n  error: No API key set. Please set the API key with --api-key or as global variable `CROWDIN_API_KEY`.\n';
+const utf8EncodingFlag = {encoding: 'utf-8'};
+
+const SUPPORTED_LOCALE = [
+  'cs',
+  'da',
+  'de',
+  'el',
+  'es',
+  'et',
+  'fi',
+  'fr',
+  'hr',
+  'hu',
+  'it',
+  'lt',
+  'nl',
+  'pl',
+  'pt',
+  'ro',
+  'ru',
+  'sk',
+  'sl',
+  'tr',
+  'uk',
+];
+
+/**
+ * @param {string} filesDir
+ * @returns {Promise<void>}
+ */
+const createEnglishFile = async filesDir => {
+  const sourceFile = path.join(filesDir, 'strings.ts');
+  const englishFile = sourceFile.replace('.ts', '-en.ts');
+
+  console.log(`Creating file "strings-en.ts" ...`);
+  await fs.copy(sourceFile, englishFile);
+
+  const content = await fs.readFile(englishFile, utf8EncodingFlag);
+  const unformattedContent = formatText(content, true);
+  const formattedContent = formatText(unformattedContent);
+  await fs.writeFile(englishFile, formattedContent, utf8EncodingFlag);
+};
+
+/**
+ * @param {string} text
+ * @param {boolean} [isTS]
+ * @returns {string}
+ */
+const formatText = (text, isTS) => {
+  const headline = "import {Supportedi18nStrings} from '../interfaces/';\n\n";
+  const identifier = 'const strings: Supportedi18nStrings = {\n';
+  const bottomline = '};\n\nexport {strings};\n';
+
+  if (isTS) {
+    const stringsRegex = new RegExp("^\\s*([^:]+):\\s('[^']+'),$", 'gm');
+    return text
+      .replace('const strings = {\n', '')
+      .replace(new RegExp(`${bottomline}\n?`), '')
+      .replace(stringsRegex, 'string.$1=$2;');
+  }
+  const stringsRegex = new RegExp("^string\\.([^=]+)=('[^']+');$", 'gm');
+
+  let strings = text.match(stringsRegex);
+
+  if (strings === null || !strings.length) {
+    return null;
+  }
+
+  strings = strings.sort();
+
+  const replacedStrings = strings.reduce((accumulator, string) => {
+    const replacedString = string.replace(stringsRegex, '$1: $2,');
+    return `${accumulator}  ${replacedString}\n`;
+  }, '');
+
+  return `${headline}${identifier}${replacedStrings}${bottomline}`;
+};
+
+/**
+ * @param {string} zipFilePath
+ * @param {string} outputDir
+ * @returns {Promise<void>}
+ */
+const unzipFiles = async (zipFilePath, outputDir) => {
+  console.log(`Reading ZIP file "${zipFilePath}" ...`);
+  const zip = new JSZip();
+  const zipFile = await fs.readFile(zipFilePath);
+  const entries = [];
+
+  const zipData = await zip.loadAsync(zipFile);
+
+  const localeRegex = new RegExp('electron/.*/strings-([a-z]{2}).js$');
```

Comment:
We need to be careful with this! On Windows directory separators are different. On my system (Windows 10) I have the following issue when executing `node trans.py download -k ...`:

```
{ [Error: ENOENT: no such file or directory, open 'D:\dev\projects\wireapp\wire-desktop\electron\src\locale\electron\locale\strings-ro.ts']
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path:
   'D:\\dev\\projects\\wireapp\\wire-desktop\\electron\\src\\locale\\electron\\locale\\strings-ro.ts' }
```

Problem is the path `wire-desktop\\electron\\src\\locale\\electron\\locale\\strings-ro.ts` which must be `wire-desktop\\electron\\src\\locale\\strings-ro.ts`